src/components/ProductContent/ProductMedia.tsx

```diff
@@ -1,16 +1,66 @@
 import React from 'react'
+import { useState, useEffect } from 'react'
+import cx from 'classnames'
+
+//TO DO: Retrieve images from database instead
+const srcs = [
+  '/images/nike-jordan-off-white.jpg',
+  '/images/nike-woman-running-shoe.jpg',
+  '/images/nike-air-vapormax-360-herren.jpg',
+]
+
+interface ISrc {
+  src: string
+}
+
+const useImage = ({ src }: ISrc) => {
+  const [loaded, setLoaded] = useState(false)
+
+  useEffect(() => {
+    const img = new Image()
+    img.src = src
+    img.onload = () => setLoaded(true)
+  }, [src])
+  return {
+    loaded,
+  }
+}
+
+const Cover = ({ src }: ISrc) => {
+  const { loaded } = useImage({ src })
+  return (
+    <img
+      className={cx('product-media__image', 'smooth', { loaded })}
+      src={src}
+    />
+  )
+}
+
+const useSrc = () => {
+  const [selectedSrc, setSrc] = useState(srcs[0])
+  return {
+    buttons: srcs.map((src) => (
+      <button
+        // TO DO: Change image location
+        style={{ background: `url(${src}) no-repeat; background-size: cover` }}
+        className={cx('product-media__button', { active: src === selectedSrc })}
+        onClick={() => setSrc(src)}
+        key={src}
+      ></button>
+    )),
+    cover: srcs.map((src) =>
+      src === selectedSrc ? <Cover key={src} src={selectedSrc} /> : null
+    ),
+  }
+}
```

Comment:
Correct me if I am wrong, you are trying to create a custom `hook` right?. If so, I don't think we need to do so since we only create a custom `hook` when we want to make it reusable for other classes. Further reading [here](https://reactjs.org/docs/hooks-custom.html#using-a-custom-hook)

I have 3 concerns here:

1. We all agreed to name a boolean variable using "is/has" as a prefix. In your code, `loaded` should be `isLoaded.`
2. Since we are using BEM CSS convention, instead of using those `inline style` (` style={{ background: url...`), we could create a CSS class instead.
3. The approach you are using is good, but I think we should use the same approach as other members. This package (`import cx from 'classnames'`) is another approach which is different from ours. You could see `ProductInformation.tsx` as an example.